### ISSUE TITLE
pylint 3.0.2 with pydantic 2.1.1: Using `Field` class shows `E1101: no-member` error

### ISSUE THREAD

**Issue by ecs-jnguyen** [pylint-dev/pylint#9189]
### Bug description

```python
"""Example pylint."""
from typing import List
from pydantic import BaseModel, Field


class Person(BaseModel):
    """Person class."""

    things: List[str] = Field(default=[], description="things")


p = Person()

# hello.py:15:0: E1101: Instance of 'FieldInfo' has no 'append' member (no-member)
p.things.append("stuff")
```

### Command used

```shell
pylint hello.py
```

### Pylint output

```python
************* Module hello
hello.py:15:0: E1101: Instance of 'FieldInfo' has no 'append' member (no-member)

------------------------------------------------------------------
Your code has been rated at 1.67/10 (previous run: 1.67/10, +0.00)
```

### Expected behavior

This is a false positive. It should not have complained about this error.

### Pylint version

```shell
pylint 3.0.2
astroid 3.0.1
Python 3.11.4 (main, Aug  9 2023, 15:20:00) [Clang 14.0.3 (clang-1403.0.22.14.1)]
```

### OS / Environment

Mac
iterm2

### Additional dependencies

```python
pydantic==2.1.1
pylint==3.0.2
```

**Comment by Timmmm** [comment]
Sounds like a duplicate of #8894

**Comment by Pierre-Sassoulas** [comment]
Thank you for triaging @Timmmm, I think you're right.
